src/dialogue choices.js

```diff
@@ -0,0 +1,226 @@
+/**
+❔
+@file dialogue choices
+@summary Present options to the user via the "say" interface, then react to their choice.
+@license MIT
+@author Violgamba (Jon Heard)
+
+
+@description
+Present options to the user via the "say" interface, then react to their choice.  This can be useful
+in a variety of ways, such as for creating dialogue trees or for picking settings.
+
+
+HOW TO USE - BASIC CHOICE RESULTS:
+1. Import this plugin into your game.
+2. Add a message event to a corner of the room.  Remove it's "one-time" field.  Set it's "say" field
+   to "hello from a location".
+3. Create a character event near the avatar.  Remove it's "say" field.  Add a "trigger-location"
+   field of location type.  Add an "after" field of javascript type.
+4. Set the "trigger-location" field of step 3 to point to the position of the event from step 2.
+5. Set the "after" field from step 3 to the following code:
+  SAY_CHOICES("Make a choice...", [
+    ["Try a function result", () => SAY("hello from a function")],
+    ["Try a string result", "hello from a string"],
+    ["Try a location result", FIELD(EVENT,"trigger-location","location")],
+    ["Try a number result", 3],
```

Comment:
```suggestion
    ["Try a number result (does nothing)", 3],
```
i'd recommend being clearer here; you note it's not supported below, but since the next step is "test the game", people probably won't read it in full until after trying it